code/modules/gamemaster/actions/atmos_leak.dm

```diff
@@ -0,0 +1,76 @@
+/datum/gm_action/atmos_leak
+	name = "atmospherics leak"
+	departments = list(ROLE_ENGINEERING, ROLE_SYNTHETIC)
+	var/area/target_area	// Chosen target area
+	var/area/target_turf	// Chosen target turf in target_area
+	var/gas_type			// Chosen gas to release
+	// Exclude these types and sub-types from targeting eligibilty
+	var/list/area/excluded = list(
+		/area/shuttle,
+		/area/crew_quarters,
+		/area/holodeck,
+		/area/engineering/engine_room
+	)
+
+	var/severity
+
+// Decide which area will be targeted!
+/datum/gm_action/atmos_leak/set_up()
+	severity = pickweight(EVENT_LEVEL_MUNDANE = 8,
+	EVENT_LEVEL_MODERATE = 5,
+	EVENT_LEVEL_MAJOR = 3
+	)
+
+	var/gas_choices = list("carbon_dioxide", "sleeping_agent") // Annoying
+	if(severity >= EVENT_LEVEL_MODERATE)
+		gas_choices += "phoron" // Dangerous
```

Comment:
Last time I checked, phoron was removed from the default air alarm filtering for `reasons`, which puts it at the same level as volatile_fuel per the comments.

Reply:
Volatile Fuel requires panic syphon, IIRC.